spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -913,8 +913,8 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     val values = Seq(BigDecimal("12345.6789"), BigDecimal("9876.5432"), BigDecimal("123.4567"))
     val df = withNulls(values)
       .toDF("b")
-      .withColumn("a", col("b").cast(DecimalType(6, 2)))
-    checkSparkAnswer(df)
+      .withColumn("a", col("b").cast(DecimalType(38, 28)))
```

Comment:
We can create the test case like
```
    val rowData = Seq(
      Row(BigDecimal("12345.6789")),
      Row(BigDecimal("9876.5432")),
      Row(BigDecimal("123.4567"))
    )
    val df = spark.createDataFrame(
      spark.sparkContext.parallelize(rowData),
      StructType(Seq(StructField("a", DataTypes.createDecimalType(10,4))))
    )
```
so that we can avoid to have `cast` in the input df